roles/translator/src/proxy/bridge.rs

```diff
@@ -497,16 +510,30 @@ impl Bridge {
         // TODO remove it sequence id is another thing
         _channel_sequence_id: u32,
         sv1_submit: Submit,
-        extranonce: Vec<u8>,
+        extranonce2: Vec<u8>,
+        version_rolling_mask: Option<HexU32Be>,
     ) -> ProxyResult<'static, SubmitSharesExtended<'static>> {
-        let version = match sv1_submit.version_bits {
-            Some(vb) => vb.0,
-            None => self
-                .channel_factory
-                .last_valid_job_version()
-                .ok_or(Error::RolesSv2Logic(RolesLogicError::NoValidJob))?,
+        let last_version = self
+            .channel_factory
+            .last_valid_job_version()
+            .ok_or(Error::RolesSv2Logic(RolesLogicError::NoValidJob))?;
+        let version = match (sv1_submit.version_bits, version_rolling_mask) {
+            (Some(vb), Some(mask)) => (last_version & !mask.0) | (vb.0 & mask.0),
```

Comment:
It would be good to add this link as a comment as to why we are doing this masking here - https://github.com/slushpool/stratumprotocol/blob/master/stratum-extensions.mediawiki#changes-in-request-miningsubmit